File: benchmark-java/src/androidTest/java/com/shinkevich/benchmark_java/PiCalculator.java
package com.shinkevich.benchmark_java;

public class PiCalculator {
    static final double START_X  = 0.0;
    static final double END_X  = 1.0;
    static double calculatePiThreadTask(int threadNum, int stepsNum, int totalThreadsNum) {
        double result = 0;
        double currentX = START_X + (END_X - START_X) / stepsNum * threadNum;
        double step =  (END_X - START_X) / stepsNum * totalThreadsNum;
        while(currentX < END_X) {
            result += 1.0 / (1.0 + currentX * currentX);
            currentX += step;
        }
        return result * (END_X - START_X) / stepsNum * 4.0;
    }

    static double calculateIntervalValue(int stepNum, int stepsNum) {
        double step = (END_X - START_X) / stepsNum;
        double currentX = START_X + stepNum * step;
        return 1.0 / (1.0 + currentX * currentX) * step * 4.0;
    }
}

File: benchmark-kotlin/src/androidTest/java/com/shinkevich/benchmark_kotlin/PiCalculator.kt
package com.shinkevich.benchmark_kotlin

object PiCalculator {
    private const val START_X = 0.0
    private const val END_X = 1.0
    fun calculatePiThreadTask(threadNum: Int, stepsNum: Int, totalThreadsNum: Int): Double {
        var result = 0.0
        var currentX = START_X + (END_X - START_X) / stepsNum * threadNum
        val step = (END_X - START_X) / stepsNum * totalThreadsNum
        while (currentX < END_X) {
            result += 1.0 / (1.0 + currentX * currentX)
            currentX += step
        }
        return result * (END_X - START_X) / stepsNum * 4.0
    }

    fun calculateIntervalValue(stepNum: Int, stepsNum: Int): Double {
        val step = (END_X - START_X) / stepsNum
        val currentX = START_X + stepNum * step
        return 1.0 / (1.0 + currentX * currentX) * step * 4.0
    }
}
